### PR TITLE
Fix #822 Add thread-safety info to AccumuloClient javadoc

### LINKED ISSUE #822: Document thread safety of AccumuloClient
The [AccumuloClient javadoc](https://github.com/apache/accumulo/blob/d3a5a45ea7530945301884bc7541adb168b05644/core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java#L32) and [client documentation](https://accumulo.apache.org/docs/2.x/getting-started/clients) provides no mention of whether or not AccumuloClient can be shared by multiple threads. It would be good to give a recommendation to users.

### PR DESCRIPTION
**Pull request by ctubbsii**

### CODE REVIEW

**Comment by milleruntime** [comment]
Not sure where the failure in Travis is from but seems unrelated.

**Comment by ctubbsii** [comment]
> Not sure where the failure in Travis is from but seems unrelated.

It is unrelated. I fixed that last night.

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java [hunk 1]
```diff
@@ -54,6 +54,13 @@
  * creating them could perform worse than Connector. Therefore, it would be better to create an
  * AccumuloClient and pass it around.
  *
+ * <p>
+ * AccumuloClient objects are intended to be thread-safe, and can be used by multiple threads.
+ * However, care should be taken to ensure that the client is eventually closed, to clean up any
+ * resources in use in the client application when all threads are finished with the AccumuloClient
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: This sounds great.  Now we need to document thread safety for the scanner and batch writer, if its not already done.

This information about closing overlaps a bit with information in the previous paragraph.

Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: > This information about closing overlaps a bit with information in the previous paragraph.

That is intentional for two reasons:
1. I wanted to have it repeated in this paragraph to expand on its brief mention in the previous paragraph, and
2. The previous paragraph may go away at some future point, and I wanted this paragraph to stand on its own.

**Pull request merged**